blackbox/docs/administration/ssl.txt

```diff
@@ -0,0 +1,114 @@
+.. _administration_ssl:
+
+================================
+Secured Communications (SSL/TLS)
+================================
+
+.. NOTE::
+
+   *Secured Communications* is an Enterprise Edition feature.
+
+Secured communications allows you to encrypt traffic between CrateDB nodes
+and clients that use PostgreSQL Wire Protocol. Connections are secured using
+Transport Layer Security (TLS).
+
+
+SSL/TLS Configuration for PostgreSQL wire protocol
+==================================================
+
+First of all you need to set the boolean setting ``ssl.psql.enabled`` to
+``true`` on each of the nodes you want to enable the secured communications.
+
+
+Configure the KeyStore
+----------------------
+
+SSL/TLS needs a keystore. The keystore holds the node certificate(s) which
+should be signed by a certificate authority (CA). A third-party CA or your
+organization's existing CA can be used. When a client connects to a node using
+SSL/TLS the client receives the certificate provided by the node and will
+determine if the node’s certificate is valid, trusted, and matches the hostname
+or IP address it is trying to connect to.
+
+.. NOTE::
+
+    Technically it's possible to disable CA checks for certificates on the
+    client. It's strongly recommended though to use certificates signed by an
+    official CA or by a private CA (company PKI) that is also known to the
+    client to be able to establish trust between nodes and clients.
+
+Once the keystore is prepared define the absolute file path of the keystore
+``.jks`` file on the node using ``ssl.keystore_filepath`` setting.
+
+.. NOTE::
+
+    Make sure that the keystore file has the right permissions and is accessible
+    by system user ``crate``.
+
+Also define the password needed to decrypt the keystore by using the
+``ssl.keystore_password`` setting.
+
+Use ``ssl.keystore_key_password`` setting to define the key password used
+when creating the keystore.
+
+A full list of the settings needed to configure SSL/TLS can
+be found :ref:`here <ssl_config>`.
+
+Configure a separate TrustStore
+-------------------------------
+
+Trusted CA certificates can also be stored in a node's keystore but optionally
+a separate truststore can be used to store them.
+
+If you want to use a separate truststore, create a node truststore and import
+the CA certificate(s) you want to trust. Once the truststore is prepared
+define the absolute file path of the truststore ``.jks`` file on the node
+using the ``ssl.truststore_filepath`` setting.
+
+.. NOTE::
+
+    Make sure that the truststore file has the right permissions and is
+    accessible by system user ``crate``.
+
+Also define the password needed to decrypt the keystore by using the
+``ssl.truststore_password`` setting.
+
+A full list of the settings needed to configure SSL/TLS can
+be found :ref:`here <ssl_config>`.
```

Comment:
reword so as not to use "here". be descriptive. like "in the x config docs" or whatever